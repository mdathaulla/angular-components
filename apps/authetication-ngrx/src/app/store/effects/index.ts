import { AuthenticationEffect } from './authentication.effect';

export const effects = [AuthenticationEffect];
