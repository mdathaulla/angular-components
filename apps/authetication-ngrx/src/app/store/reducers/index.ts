import { ActionReducerMap } from "@ngrx/store";
import * as fromAuthentication from "../../authentication/store";
export interface State {
  authentication: fromAuthentication.AuthenticationState;
}

export const reducers: ActionReducerMap<State> = {
  authentication: fromAuthentication.reducer
};


