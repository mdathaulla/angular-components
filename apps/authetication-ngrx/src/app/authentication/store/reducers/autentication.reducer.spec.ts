import * as fromStore from "../../store";
import { AuthenticationData } from "../../../core/interfaces/authentication-data.interface";
import { User } from "../../../core/interfaces/user.interface";

describe("Authentication reducer", () => {
  it("should return initial state when its initalized", () => {
    const action = { type: "@@init" };
    const state = fromStore.reducer(undefined, action);
    expect(state).toBe(fromStore.initialState);
  });
  it("should load user and token when api is successfull login", () => {
    const token = `agf11231asd`;
    const user: User = {
      id: `1`,
      name: "Pedro Benjamin Lescano Pasquet",
      username: "plescanopasquet"
    };
    const payload: AuthenticationData = { user, token };
    const action = fromStore.loginSuccess({ payload });
    const state = fromStore.reducer(fromStore.initialState, action);
    expect(state).toMatchSnapshot();
  });
  it("should load user and token if exist in local", () => {
    const token = `agf11231asd`;
    const user: User = {
      id: `1`,
      name: "Pedro Benjamin Lescano Pasquet",
      username: "plescanopasquet"
    };
    const payload: AuthenticationData = { user, token };
    const action = fromStore.getAuthenticationLocal({ payload });
    const state = fromStore.reducer(fromStore.initialState, action);
    expect(state).toMatchSnapshot();
  });

  it("should clean user and token if logout success", () => {
    const token = `agf11231asd`;
    const user: User = {
      id: `1`,
      name: "Pedro Benjamin Lescano Pasquet",
      username: "plescanopasquet"
    };
    const payload: AuthenticationData = { user, token };
    const authenticatedAction = fromStore.getAuthenticationLocal({ payload });
    const authenticatedState = fromStore.reducer(
      fromStore.initialState,
      authenticatedAction
    );
    const logoutAction = fromStore.logoutSuccess();
    const state = fromStore.reducer(authenticatedState, logoutAction);
    expect(state).toMatchSnapshot();
  });
});
