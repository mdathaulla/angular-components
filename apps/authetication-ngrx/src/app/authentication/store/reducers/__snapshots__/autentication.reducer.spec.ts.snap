// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication reducer should clean user and token if logout success 1`] = `
Object {
  "loaded": false,
  "loading": false,
  "token": null,
  "user": null,
}
`;

exports[`Authentication reducer should load user and token if exist in local 1`] = `
Object {
  "loaded": true,
  "loading": false,
  "token": "agf11231asd",
  "user": Object {
    "id": "1",
    "name": "Pedro Benjamin Lescano Pasquet",
    "username": "plescanopasquet",
  },
}
`;

exports[`Authentication reducer should load user and token when api is successfull login 1`] = `
Object {
  "loaded": true,
  "loading": false,
  "token": "agf11231asd",
  "user": Object {
    "id": "1",
    "name": "Pedro Benjamin Lescano Pasquet",
    "username": "plescanopasquet",
  },
}
`;
